sycl/doc/extensions/experimental/sycl_ext_oneapi_enqueue_custom_operation.asciidoc

```diff
@@ -0,0 +1,203 @@
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for GPU devices and
+only when using the CUDA or HIP backends.  Attempting to use this extension in
+kernels that run on other devices or backends may result in undefined
+behavior.  Be aware that the compiler is not able to issue a diagnostic to
+warn you if this happens.
+
+
+== Overview
+
+This extension is derived from an experimental AdaptiveCpp extension, which is
+documented
+https://github.com/AdaptiveCpp/AdaptiveCpp/blob/develop/doc/enqueue-custom-operation.md[here]._
+
+The extension aims to integrate asynchronous interop work submitted through the
+backend API with the asynchrony of the SYCL runtime DAG. As such it is required
+that any interop calls used within `ext_oneapi_enqueue_custom_operation`
+are asynchronous.
+
+```c++
+sycl::queue q;
+
+q.submit([&](sycl::handler &cgh) {
+    sycl::accessor acc{buf, cgh};
+
+    cgh.ext_oneapi_enqueue_custom_operation([=](sycl::interop_handle &h) {
+      // Can extract device pointers from accessors
+      void *native_mem = h.get_native_mem<sycl::backend::hip>(acc);
+      // Can extract stream (note: get_native_queue() may not be 
+      // supported on CPU backends)
+      hipStream_t stream = h.get_native_queue<sycl::backend::hip>();
+      // Can extract HIP device (note: get_native_device() may not be
+      // supported on CPU backends)
+      int dev = h.get_native_device<sycl::backend::hip>();
+      // Can enqueue arbitrary backend operations. This could also be a kernel
+      // launch or call to a library that enqueues operations on the stream etc
+      hipMemcpyAsync(target_ptr, native_mem, test_size * sizeof(int),
+                      hipMemcpyDeviceToHost, stream);
+    });
+  });
+q.wait();
+```
+
+Provided that the correct dependencies are handled in the SYCL runtime, there
+is no need to do any further synchronization within the backend API in the user
+submitted lambda.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_ENQUEUE_CUSTOM_OPERATION` to one of the values defined
+in the table below.  Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's features the
+implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+
+=== Synchronization
+
+The goal of `ext_oneapi_enqueue_custom_operation` is to integrate interop
+work within the asynchronous SYCL DAG. As such, the user defined lambda must
```

Comment:
I would re-phrase this as integrate with the DAG creation, to more clearly differentiate it with the existing host task inteorp.

Reply:
Ok sure, will reword.